Snowflake.Data/Core/FileTransfer/StorageClient/SFS3Client.cs

```diff
@@ -176,27 +148,46 @@ public RemoteLocation ExtractBucketNameAndPath(string stageLocation)
         /// <returns>The file header of the S3 file.</returns>
         public FileHeader GetFileHeader(SFFileMetadata fileMetadata)
         {
-            // Get the client
-            SFS3Client SFS3Client = (SFS3Client)fileMetadata.client;
-            AmazonS3Client client = SFS3Client.S3Client;
+            Uri uri = GetFileUri(fileMetadata);
+
+            // Use empty string hash for content
+            SortedDictionary<string, string> headers = new SortedDictionary<string, string>
+            {
+                {X_AMZ_CONTENT_SHA256, EMPTY_BODY_SHA256},
+                {X_AMZ_SECURITY_TOKEN, fileMetadata.stageInfo.stageCredentials[AWS_TOKEN]},
+                {CONTENT_TYPE, HTTP_HEADER_VALUE_OCTET_STREAM}
+            };
+
+            string authorization = Sign(
+                fileMetadata.stageInfo, 
+                EMPTY_BODY_SHA256, 
+                "GET", 
+                uri, 
+                headers);
 
-            GetObjectRequest request = GetFileHeaderRequest(ref client, fileMetadata);
-            GetObjectResponse response = null;
             try
             {
-                // Issue the GET request
-                var task = client.GetObjectAsync(request);
-                task.Wait();
+                // Send the request
+                HttpWebRequest request = (HttpWebRequest)HttpWebRequest.Create(uri.AbsoluteUri);
```

Comment:
Previously it's an async operation. Are we getting same behavior after the change?

Reply:
Previously the client.GetObjectAsync() function was used but task.Wait() was synchronously blocking it. So the behaviour is similar since they both execute as synchronous